### ISSUE TITLE
Multiple account support

### ISSUE THREAD

**Issue by zenMaya** [michaeladler/notmuch-mailmover#116]
First, let me thank you for your great program! It is extremely useful.

However I have a proposal for you. I understand that this would probably require some architectural changes. But the problem is as follows.

I have two mail accounts, let's call them `foo@baz.org` and `bar@baz.org`. Both of them are in the `~/Mail` directory. I want to move and sync their tags and folders simply `junk -> Junk`, `deleted -> Trash`, `sent -> Sent` etc. But currently my setup must do all the tag synchronization manually:

```yaml
  - folder: foo/Trash
    query: (from:foo@baz.org or to:foo@baz.org) and tag:deleted

  - folder: foo/Sent
    query: (from:foo@baz.org or to:foo@baz.org) and tag:sent and not tag:deleted

  - folder: foo/Archive
    query: (from:foo@baz.org or to:foo@baz.org) and tag:archive and not tag:sent and not tag:deleted

  - folder: foo/Drafts
    query: (from:foo@baz.org or to:foo@baz.org) and tag:draft and not tag:deleted and not tag:archive

  - folder: foo/Junk
    query: (from:foo@baz.org or to:foo@baz.org) and tag:junk and not tag:deleted and not tag:archive

  - folder: bar/Trash
    query: (from:bar@baz.org or to:bar@baz.org) and tag:deleted

  - folder: bar/Sent
    query: (from:bar@baz.org or to:bar@baz.org) and tag:sent and not tag:deleted

  - folder: bar/Archive
    query: (from:bar@baz.org or to:bar@baz.org) and tag:archive and not tag:sent and not tag:deleted

  - folder: bar/Drafts
    query: (from:bar@baz.org or to:bar@baz.org) and tag:draft and not tag:deleted and not tag:archive

  - folder: bar/Junk
    query: (from:bar@baz.org or to:bar@baz.org) and tag:junk and not tag:deleted and not tag:archive
```
    
I know that this is not a large problem, and I easily fixed it by some macro editing. But as a in the future proposal, I think it has some merit.

PS. you don't need to implement this. I am perfectly fine using the program as is, and thank you for making it.

**Comment by michaeladler** [comment]
Hi @zenMaya,

thanks for your feedback. Personally I use a dedicated notmuch database for each mail account. In that setup, I could change the type of `notmuch_config` from `string` to `[string]`, e.g.

```yaml
notmuch_config:
  - ~/.config/notmuch/foo/config
  - ~/.config/notmuch/bar/config
```

Then `notmuch-mailmover` would just loop over the notmuch configurations and apply the rules. 

For your setup, I believe the most generic solution would be to write the configuration file in Lua and use a crate like `mlua` to load it into notmuch-mailmover. This approach would allow you to iterate over the accounts in your Lua configuration and generate rules dynamically. What do you think? Do you also have a solution in mind?

P.S. I’ve been considering offering another strategy for applying rules: I want to provide the option to apply the first matching rule and then stop processing further. This would allow for overlapping rules and enable more concise rule lists.

**Comment by michaeladler** [comment]
Considering this resolved via b28a3d843b870bcebe0e530a70e3d6d62d2016a8 and in partciular https://github.com/michaeladler/notmuch-mailmover/blob/1323e7e62b75a318213b3c5ebf2a6bf6b08beebd/example/config_multiple_accounts.lua

Feel free to report back any feedback you might have :)
